src/presign/participant.rs

```diff
@@ -85,6 +85,35 @@ pub enum Status {
     TerminatedSuccessfully,
 }
 
+/// PresignContext includes shared parameters in
+/// [`SharedContext`] and public parameters as
+/// defined in [`AuxInfoPublic`].
+pub struct PresignContext {
+    shared_context: SharedContext,
+    auxinfo_public: Vec<AuxInfoPublic>,
+}
+
+impl ProofContext for PresignContext {
+    fn as_bytes(&self) -> Result<Vec<u8>> {
+        Ok([
+            self.shared_context.as_bytes()?,
+            bincode::serialize(&self.auxinfo_public)
+                .map_err(|_| InternalError::InternalInvariantFailed)?,
+        ]
+        .concat())
+    }
+}
+
+impl PresignContext {
+    pub(crate) fn collect(mut auxinfo_public: Vec<AuxInfoPublic>, p: &PresignParticipant) -> Self {
```

Comment:
suggestion: Since the vector you pass here comes from `self`, consider not passing it as a parameter, and putting the `self.all_auxinfo_publics()` function call in here.

Reply:
good catch